src/schema/city/index.ts

```diff
@@ -118,3 +138,27 @@ const lookupCity = slug => {
   }
   return cityData[slug]
 }
+
+const nearestCityOrNull = latLng => {
+  const orderedCities = citiesOrderedByDistance(latLng)
+  const closestCity = orderedCities[0]
+
+  if (isCloseEnough(latLng, closestCity)) {
+    return closestCity
+  }
+  return null
+}
+
+const citiesOrderedByDistance = latLng => {
+  let cities = Object.values(cityData)
+  cities.sort((a: any, b: any) => {
```

Comment:
Tried to avoid `any`, but typing a comparator function that reaches into its args was beyond me in this PR.

Reply:
Oops. Wrong button, didn’t mean to resolve but also don’t see an undo option (on my phone).

I wanted to say that I would expect these params to get typed based on `cities`. Isn’t that typed as `string[]`?

Reply:
`cities` here is an array of objects, shaped [like this](https://github.com/artsy/metaphysics/blob/876d7346a9da828397850094b0fee10f85e5a3fc/src/schema/city/city_data.json#L3-L5).

Without the `any` I got tsc errors like this…

<img width="802" alt="err" src="https://user-images.githubusercontent.com/140521/50495407-f5166080-09f6-11e9-867e-8b71f2a00026.png">

Then if I tried to devise a type that has coordinates within it (excuse the goofy naming here) I got something else…

<img width="704" alt="err2" src="https://user-images.githubusercontent.com/140521/50495430-0eb7a800-09f7-11e9-88b8-c63ef8b1fd2e.png">

That one left me stumped.

Reply:
Ah I see; ofc, I don’t know why I was thinking it would be a string. I don’t recall if `Object.values()` is able to infer the type of the objects. Is the `cityData` typed as you’d expect? If not, then TS isn’t making use of a new compiler setting that is able to infer types from JSON files.

Reply:
FYI The type error you get in the second case is, I think, TS complaining about you passing a function that takes `HasCoordinates` parameters to a function that will yield `{}` (empty object).

Reply:
I think you had the right idea with type inference not working with `Object.values()` — which makes sense since the values could be totally heterogeneous.

Typing that assignment explicitly [here](https://github.com/artsy/metaphysics/pull/1448/files#diff-36aa4512b29ee50fae45b7352a1c9cd6R153) was the key to making this work. Thanks!